File: backend/src/admin/dto/cabinet.info.response.dto.ts
import { ApiPropertyOptional } from '@nestjs/swagger';
import { CabinetDto } from './cabinet.dto';
import { LentDto } from './lent.dto';

/**
 * 사물함에 대한 정보와 대여한 사람들 정보
 * @extends CabinetExtendDto
 */
export class CabinetInfoResponseDto extends CabinetDto {
  @ApiPropertyOptional({
    description: '대여되어 있을 경우 대여 정보',
    type: [LentDto],
  })
  lent_info?: LentDto[]; // 대여 정보 (optional)
}

File: backend/src/main/java/org/ftclub/cabinet/dto/response/CabinetInfoResponseDto.java
package org.ftclub.cabinet.dto.response;

import lombok.*

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CabinetInfoResponseDto extends CabinetDto {
    /**
     * FIXME:
     * 위치 정보 수정
     */
    private String building;
    private Integer floor;
    private Optional<LentDto[]> lentInfo; // Optional 처리를 따로 해야하는지?
}